### PR TITLE
[Keymap] Add Engram Planck keymap

### PR DESCRIPTION
**Pull request by CallumAndrew**
## Description

Initial commit for the keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/planck/keymaps/engram/config.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
+
```
Comment (fauxpark) [inline keyboards/planck/keymaps/engram/config.h:47]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/planck/keymaps/engram/config.h (hunk 1 above)
Comment (CallumAndrew) [reply keyboards/planck/keymaps/engram/config.h:47]: Aren't they the same?

Comment (fauxpark) [reply keyboards/planck/keymaps/engram/config.h:47]: What is "they"?

Comment (CallumAndrew) [reply keyboards/planck/keymaps/engram/config.h:47]: The original config.h file and the suggested change

Comment (fauxpark) [reply keyboards/planck/keymaps/engram/config.h:47]: The suggestion is to remove the lines. That's why they are in red.

**Pull request closed without merging**